polypomo

```diff
@@ -133,6 +138,33 @@ class Status:
             self.timer = Timer(self.worktime)
 
 
+@contextmanager
+def setup_conn(path):
+    # check if the database exist
+    conn = sqlite3.connect(path)
+    cur = conn.cursor()
+    cur.execute("""CREATE TABLE IF NOT EXISTS sessions (
+                        date text,
+                        start text,
+                        stop text
+        )""")
+    try:
+        yield cur
+    finally:
+        conn.commit()
+        conn.close()
+
+def db_save(status, path, active):
+    if status == "work":
+        with setup_conn(path) as conn:
+            if  active:
+                conn.execute(f"INSERT INTO sessions VALUES (date('now'), time('now'), NULL)")
+            else:
+                conn.execute(f"""UPDATE sessions
+                                 SET stop = time('now')
+                                 WHERE stop IS NULL
```

Comment:
What if polypomo crashes and there's an incomplete entry from the past?
They will all have the same stop timestamp. Can we maybe order by start time and `LIMIT 1` so this only affects the most recent record?